### PR TITLE
Output the commands used with genmeshbox in a separate log file

### PR DESCRIPTION
**Pull request by vbaconnet**
Added a log that shows which parameters were used with genmeshbox, to have a trace.

### CODE REVIEW

**Comment by timofeymukha** [comment]
Naive question: would it be easier / better to print things into stdout by default, and then if the user wants it logged the standard linux output redirection can be used?

**Comment by timfelle** [comment]
> Naive question: would it be easier / better to print things into stdout by default, and then if the user wants it logged the standard linux output redirection can be used?

The benefit of this approach is that one could separate run time messages from the parameters more easily.

Just a little fever dream:
A cool feature based on this could be to dump it as a json, which could then be reinserted into genmeshbox. This could allow us to recreate the exact mesh and the json could then follow the dataset so you dont have to remember how you generated it.

Once that is done, you could even put the specification into the casefile and have the neko driver call genmeshbox if the mesh specifications are there, but the mesh is not.

**Comment by vbaconnet** [comment]
> Just a little fever dream: A cool feature based on this could be to dump it as a json, which could then be reinserted into genmeshbox. This could allow us to recreate the exact mesh and the json could then follow the dataset so you dont have to remember how you generated it.
>
> Once that is done, you could even put the specification into the casefile and have the neko driver call genmeshbox if the mesh specifications are there, but the mesh is not.


That does sound really nice :) Unfortunately I am not so well-versed with the json fortran stuff and perhaps this will be a  bit more work than I can do for now since this was supposed to be a quick fix... But I agree it would be super nice. 

I would suggest we add the json stuff as todo and either merge or we close this PR :)

**Review by njansson — APPROVED** [review]

File: contrib/genmeshbox/genmeshbox.f90 [hunk 1]
```diff
@@ -67,6 +72,43 @@ program genmeshbox
...
+        ! files with genmeshbox.log.1, .2, .3, etc
+        if (i .lt. 10) then
+           write(log_fname,'(A,I1)') "genmeshbox.log.", i
+        else if (i .lt. 100) then
+           write(log_fname,'(A,I2)') "genmeshbox.log.", i
+        else if (i .lt. 1000) then
+           write(log_fname,'(A,I3)') "genmeshbox.log.", i
+        else
+           write(log_fname,'(A,I4)') "genmeshbox.log.", i
+        end if
```
Comment (timfelle) [inline contrib/genmeshbox/genmeshbox.f90:107]: Could we maybe format it as `genmeshbox_N.log` instead to avoid modifying the file extension.

Comment (vbaconnet) [reply contrib/genmeshbox/genmeshbox.f90:107]: sure!

**Review by MartinKarp — COMMENTED** [review]
Ok to merge this one before before #1407 @adperezm?

**Comment by adperezm** [comment]
> Ok to merge this one before before #1407 @adperezm?

Yeah, that sounds good

**Review by MartinKarp — APPROVED** [review]

**Pull request merged**